packages/zend-xml/library/Zend/Xml/Security.php

```diff
@@ -45,6 +45,19 @@ protected static function heuristicScan($xml)
         }
     }
 
+    /**
+     * Wrapper for libxml_disable_entity_loader which is deprecated in PHP 8. libxml 2.9.0 disables external entity
+     * loading by default. PHP 8.0 has deprecated libxml_disable_entity_loader() method and it requires libxml 2.9.0+.
+     * @see https://github.com/php/php-src/pull/5867
+     *
+     * @param bool $disable
+     * @return bool the previous value (true = disabled)
+     */
+    protected static function disableEntityLoader($disable = true)
+    {
+        return LIBXML_VERSION < 20900 ? libxml_disable_entity_loader($disable) : true;
```

Comment:
1. are you sure you want to always return true, even if you intend to disable? seems you want to return `!$disable`, so the api would behave like the change "worked".
3. make method private, makes maintenance easier if things not exported too much
2. the return value is not used at all. so make method void?

Reply:
EDIT: the return value **is** used, mislooked the usage. and as the method is private does not matter what is returned in the "no-op" mode.

Reply:
1. There's something to be said for that, the toggling still works even on PHP8 with the deprecated warning. As you said it doesn't matter here but I might as well do it 'right'.
2.  I went with the standard in that file. Though I doubt there's code out there extending that Security class and wanting to use this method so private is also fine with me.

Should I make both changes?

Reply:
make it only `private`.

Reply:
Done.